gettsim/taxes/zu_verst_eink/freibeträge.py

```diff
@@ -0,0 +1,239 @@
+        )
+    )
+
+    return pd.Series(data=out, index=behinderungsgrad.index)
+
+
+def _hh_freib_bis_2014(alleinerziehend, eink_st_abzuege_params):
+    """
+    Calculates tax reduction for single parents. Used to be called
+    'Haushaltsfreibetrag'
+
+    Parameters
+    ----------
+    alleinerziehend
+    eink_st_abzuege_params
+
+    Returns
+    -------
+
+    """
+    out = alleinerziehend.astype(float) * 0
+    out.loc[alleinerziehend] = eink_st_abzuege_params["alleinerziehenden_freibetrag"]
+    return out
+
+
+def _hh_freib_seit_2015(alleinerziehend, _anz_kinder_in_tu, eink_st_abzuege_params):
+    """
+    Calculates tax reduction for single parents. Since 2015, it increases with
+    number of children. Used to be called 'Haushaltsfreibetrag'
+
+    Parameters
+    ----------
+    alleinerziehend
+    kind
+    tu_id
+    eink_st_abzuege_params
+
+    Returns
+    -------
+
+    """
+    out = alleinerziehend.astype(float) * 0
+    out.loc[alleinerziehend] = (
+        eink_st_abzuege_params["alleinerziehenden_freibetrag"]
+        + _anz_kinder_in_tu.loc[alleinerziehend]
+        * eink_st_abzuege_params["alleinerziehenden_freibetrag_zusatz"]
+    )
+    return out
+
+
+def altersfreib(
+    bruttolohn_m,
+    alter,
+    kapital_eink_m,
+    eink_selbstst_m,
+    vermiet_eink_m,
+    eink_st_abzuege_params,
+):
+    """
+    Calculates the deductions for elderly. Not tested yet!
+
+    Parameters
+    ----------
+    bruttolohn_m
+    alter
+    kapital_eink_m
+    eink_selbstst_m
+    vermiet_eink_m
+    eink_st_abzuege_params
+
+    Returns
+    -------
+
+    """
+    out = bruttolohn_m * 0
+    out.loc[alter > 64] = (
+        eink_st_abzuege_params["altersentlastung_quote"]
+        * 12
+        * (
+            bruttolohn_m
+            + (kapital_eink_m + eink_selbstst_m + vermiet_eink_m).clip(lower=0)
+        )
+    ).clip(upper=eink_st_abzuege_params["altersentlastungsbetrag_max"])
+    return out
+
+
+def _sonderausgaben_bis_2011(kind, eink_st_abzuege_params):
+    """
+    Until 2011, we just use the lumpsum payment.
+    Parameters
+    ----------
+    kind
+    eink_st_abzuege_params
+
+    Returns
+    -------
+
+    """
+    out = kind.astype(float) * 0
+    out.loc[~kind] = eink_st_abzuege_params["sonderausgabenpauschbetrag"]
+    return out
+
+
+def _sonderausgaben_ab_2012(
+    betreuungskost_m, tu_id, kind, anz_erwachsene_in_tu, eink_st_abzuege_params
+):
+    """
+    Calculating sonderausgaben for childcare. We follow 10 Abs.1 Nr. 5 EStG. You can
+    details here https://www.buzer.de/s1.htm?a=10&g=estg.
+    Parameters
+    ----------
+    betreuungskost_m
+    tu_id
+    kind
+    eink_st_abzuege_params
+    anz_erwachsene_in_tu
+
+    Returns
+    -------
+
+    """
+    abziehbare_betreuungskosten = (12 * betreuungskost_m).clip(
+        upper=eink_st_abzuege_params["kinderbetreuungskosten_abz_maximum"]
+    )
+
+    berechtigte_kinder = (kind.astype(int)).groupby(tu_id).transform(sum)
```

Comment:
If it is not strictly necessary, we should avoid typing because it might create copies which is expensive.
```suggestion
    berechtigte_kinder = kind.groupby(tu_id).transform(sum)
```